packages/@aws-cdk/aws-servicecatalog/lib/private/association-manager.ts

```diff
@@ -28,8 +29,60 @@ export class AssociationManager {
     };
   }
 
+  public static constrainTagUpdates(portfolio: IPortfolio, product: IProduct, options: TagUpdateConstraintOptions): void {
+    this.validateConstraintDescription(this.prettyPrintAssociation(portfolio, product), options);
+    const association = this.associateProductWithPortfolio(portfolio, product);
+    const constructId = `ResourceUpdateConstraint${association.associationKey}`;
+
+    if (!portfolio.node.tryFindChild(constructId)) {
+      const constraint = new CfnResourceUpdateConstraint(portfolio as unknown as cdk.Resource, constructId, {
+        acceptLanguage: options.messageLanguage,
+        description: options.description,
+        portfolioId: portfolio.portfolioId,
+        productId: product.productId,
+        tagUpdateOnProvisionedProduct: options.allow === false ? 'NOT_ALLOWED' : 'ALLOWED',
+      });
+
+      // Add dependsOn to force proper order in deployment.
+      constraint.addDependsOn(association.cfnPortfolioProductAssociation);
+    } else {
+      throw new Error(`Cannot have multiple tag update constraints for association ${this.prettyPrintAssociation(portfolio, product)}`);
+    }
+  }
+
+  public static deployWithStackSets(portfolio: IPortfolio, product: IProduct, accounts: string[], regions: string[], adminRole: iam.IRole,
+    executionRole: iam.IRole, allowInstanceControl: boolean, options: CommonConstraintOptions) {
+
+    this.validateConstraintDescription(this.prettyPrintAssociation(portfolio, product), options);
+    const association = this.associateProductWithPortfolio(portfolio, product);
+    // Check if a launch role has already been set.
+    if (portfolio.node.tryFindChild(`LaunchRoleConstraint${association.associationKey}`)) {
+      throw new Error(`Cannot configure stackset deployment when a launch role is already defined for association ${this.prettyPrintAssociation(portfolio, product)}`);
+    }
+
+    const constructId = `StackSetConstraint${association.associationKey}`;
+    if (!portfolio.node.tryFindChild(constructId)) {
+      const constraint = new CfnStackSetConstraint(portfolio as unknown as cdk.Resource, constructId, {
+        acceptLanguage: options.messageLanguage,
+        description: options.description ?? '',
```

Comment:
Hmm. Is this description randomly required? Is it not in the other constraints? Is an empty string allowed here?

Reply:
There is issue with it being required for this constraint, but that is going be fixed and removed.  We don't need to force users to do it in the meantime.